ibm_cloud_sdk_core/authenticators/basic_authenticator.py

```diff
@@ -44,9 +58,16 @@ def validate(self):
                 'The username and password shouldn\'t start or end with curly brackets or quotes. '
                 'Please remove any surrounding {, }, or \" characters.')
 
-    def authenticate(self, req):
-        """
-        Adds the Authorization header, if applicable
+    def authenticate(self, req: Request):
+        """Add basic authentication information to a request.
+
+        Basic Authorization will be added to the request's headers in the form:
+
+            Authorization: Basic <encoded username and password>
+
+        Args:
+            req: The request to add basic auth information too. Must contain a key to a dictionary
+            called headers.
         """
         authstring = "{0}:{1}".format(self.username, self.password)
```

Comment:
This might be a little beyond the scope of your task, but there's really no reason why we can't cache the Authorization header value (i.e. update it any time the username/password is set or updated), then just use the cached value inside the `authenticate()` method.

Reply:
👍